### DISCUSSION TITLE
start the streaming from a specific seconds

### DISCUSSION THREAD

**Discussion by hamedbrd** [pion/webrtc#2148] [Q&A] [answered]
Hi guys,

Recently I have been using this snippet [https://github.com/pion/webrtc/tree/master/examples/play-from-disk](https://github.com/pion/webrtc/tree/master/examples/play-from-disk) to stream some audio files. as I am so new on this part, I got these two issues:

first, I am trying to find out how I can start the streaming from a specific second. 
and second, I am not sure how I should implement the pause function.

I would really appreciate it if you give me some tips on these issues I have.

**Comment by Sean-Der** [comment, marked as answer]
Hi @hamedbrd 

> first, I am trying to find out how I can start the streaming from a specific second.

In play-from-disk [here](https://github.com/pion/webrtc/blob/master/examples/play-from-disk/main.go#L182) we loop the file and compute the duration of each ogg page. I would start with discarding samples until it reaches the time you want to start playing.

For pause you could Stop/Sleep in the same loop.

If you want to start a repo with edits I am happy to review any changes you make!

**Comment by hamedbrd** [comment]
Thanks, @Sean-Der  for your answer

I am handling it by discarding samples like this now 
```
for i := 0; i < (int(req.Second)*1000)/millisecond; i++ {
			_, _, oggErr := ogg.ParseNextPage()
			if oggErr == io.EOF {
				g.logger.Info("All audio pages parsed and sent")
				break
			}
		}
```
just before this [line](https://github.com/pion/webrtc/blob/master/examples/play-from-disk/main.go#L168)

but I am not sure this is a proper way to handle this. how about if we decide to jump to the second 10 from the second 40?

**Reply by Sean-Der** [reply]
Hi @hamedbrd 

Can you put this in a gist or repo? I don't believe what you are doing is correct.

You need to get the duration of each page like so

```go
totalDuration := time.Millisecond * 0

for  {
    pageData, pageHeader, err := ogg.ParseNextPage()
    if err != nil {
        panic(err)
    }

    sampleCount := float64(pageHeader.GranulePosition - lastGranule)
    lastGranule = pageHeader.GranulePosition
    totalDuration += time.Duration((sampleCount/48000)*1000) * time.Millisecond
    if totalDuration >= 40 * time.Second {
        // You have read 40 seconds of data
    }
}
